apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -0,0 +1,374 @@
+/* global window */
+
+/**
+ * Teacher Application questionaire
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import {Radio, Checkbox, FormControl, FormGroup, ControlLabel} from 'react-bootstrap';
+
+function FieldGroup({ id, label, ...props }) {
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+    </FormGroup>
+  );
+}
+
+const grades = ['Kindergarten'].concat(_.range(1,13));
+const subjects = ['Computer Science', 'Computer Literacy', 'Math', 'Science', 'History', 'English', 'Music', 'Art', 'Multimedia', 'Foreign Language'];
+const yesNoResposes = ['Yes', 'No'];
+
+const TeacherApplication = React.createClass({
+
+  getInitialState() {
+    return {
+      selectedCourse: ''
+    };
+  },
+
+  buttonList(type, label, groupName, answers, allowOther = false) {
+    const options = answers.map( (answer, i) => {
+      if (type === 'radio') {
+        return (
+          <Radio
+            value={answer}
+            label={answer}
+            key={i}
+            name={groupName}
+          >
+            {answer}
+          </Radio>
+        );
+      } else if (type === 'check') {
+        return (
+          <Checkbox
+            value={answer}
+            label={answer}
+            key={i}
+            name={groupName}
+          >
+            {answer}
+          </Checkbox>
+        );
+      }
+    });
+
+    if (allowOther) {
+      const otherDiv = (
+        <div style={{display: 'inline'}}>
+          <span style={{marginRight: '5px'}}>
+            Other:
+          </span>
+          <input type="text" id={groupName + 'Other'}/>
```

Comment:
Does this render the text box inline with the radio buttons / checkboxes? Slick.